lib/atomic_time/core_ext/active_record/base.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'active_record/base'
+require 'active_support/duration'
+
+module ActiveRecord
+  class Base
+    class << self
+      def time_interval(lower, upper)
+        lower, upper = [lower, upper].map(&:to_sym)
+
+        before_validation do
+          lower_val, upper_val = [lower, upper].map { |key| send key }
+          adj = upper_val.is_a?(DateTime) ? 1 : ActiveSupport::Duration::SECONDS_PER_DAY
```

Comment:
Yeah this case handling is probably unnecessary. I threw it in here just cause I thought of it and it was relatively easy to solve.

I'm not sure what you're getting at with using `acts_like_time?` though. In this gem `acts_like_time?` will return true for `DateTime` `Time` and `AtomicTime` and the purpose of this conditional is because `DateTime#+` interprets integer arguments as days, while `Time#+` interprets them as seconds.

Reply:
Ah, I completely misunderstood the purpose of this line of code.

I assumed it was, "if it's time-y, add seconds, if it's date-y add days" hence my comment below too.

I think perhaps add a comment explaining the difference in `:+` behaviors, than?